x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -57,6 +57,7 @@ export const enhancedEsSearchStrategyProvider = (
           utils.toSnakeCase({
             ...(await getDefaultSearchParams(uiSettingsClient)),
             batchedReduceSize: 64,
+            keepOnCompletion: true,
```

Comment:
curious what's this and why we need this now

Reply:
This ensures that we always receive an `id` in the response from the `_async_search` request. If we don't send this, then if a search request finishes before the `wait_for_completion_timeout` value, we won't get an ID that we can store in the background session ID mapping.